File: src/main/java/common/block/BlockOre.java
package common.block;

import common.item.Item;
import common.registries.Items;
import common.world.World;

import java.awt.*;

public class BlockOre extends BlockColored {
    private Item drop;
    private int color = 0xFFFFFF;

    public BlockOre(String name, Item drop, int color) {
        super(name);
        this.drop = drop;
        this.color = color;
    }

    public BlockOre(String name, Item drop) {
        super(name);
        this.drop = drop;
    }

    public BlockOre(String name, int color) {
        super(name);
        this.color = color;
    }

    public BlockOre(String name) {
        super(name);
    }

    @Override
    public void onBlockRightClick() { }

    @Override
    public void onBlockCollision() { }

    @Override
    public void onTick(World world, int x, int y) { }

    public Item getBlockDrop() {
        if (drop == null){
            return this.drop =Items.getItem(this.getName());
        }
        return drop;
    }

    public Color getColor() {
        return  new Color(color);
    }
}

File: src/main/java/common/block/BlockOre.kt
package common.block

import common.item.Item
import common.registries.ItemRegistry
import java.awt.Color

class BlockOre : BlockColored {
    private var drop: Item? = null
    private var color = 0xFFFFFF

    val blockDrop: Item
        get() = if (drop != null) {
            this.drop!!
        } else ItemRegistry.getItem(this.name)

    constructor(name: String, drop: Item, color: Int) : super(name) {
        this.drop = drop
        this.color = color
    }

    constructor(name: String, drop: Item) : super(name) {
        this.drop = drop
    }

    constructor(name: String, color: Int) : super(name) {
        this.color = color
    }

    constructor(name: String) : super(name) {}

    fun getColor(): Color {
        return Color(color)
    }
}
